fuse_publishers/include/fuse_publishers/graph_publisher.h

```diff
@@ -0,0 +1,192 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2019, Clearpath Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_PUBLISHERS_GRAPH_PUBLISHER_H
+#define FUSE_PUBLISHERS_GRAPH_PUBLISHER_H
+
+#include <fuse_core/async_publisher.h>
+#include <fuse_core/graph.h>
+#include <fuse_core/macros.h>
+#include <fuse_core/transaction.h>
+#include <fuse_core/uuid.h>
+
+#include <fuse_msgs/Graph.h>
+
+#include <ros/ros.h>
+
+#include <unordered_map>
+
+
+namespace fuse_publishers
+{
+
+/**
+ * @brief Publisher plugin that publishes the latest graph
+ *
+ * There are two options:
+ *  - The graph is only published when variables or constraints are added or removed.
+ *  - The graph is also published on every optimization update, even if the graph topology didn't change.
+ *    Remember the variables' solution might have changed on every graph optimization cycle.
+ *
+ * Parameters:
+ *  - device_id (uuid string, default: 00000000-0000-0000-0000-000000000000) The device/robot ID to publish
+ *  - device_name (string) Used to generate the device/robot ID if the device_id is not provided
+ *  - publish_optimized_graph (bool, default: true) Flag indicating that the graph should be always published, i.e. not
```

Comment:
With my setup, the inputs frequency is quite high, so I never got a graph with an empty transaction, which is what I mean by optimized graph (only the variables are optimized out, but the variables and constraints in the graph are the same, as well as their topology).

Let me know what do you think of this option?